third_party/move/move-compiler-v2/tests/ability-transform/destroy_after_call.exp

```diff
@@ -14,12 +14,14 @@ fun m::g() {
      var $t1: &mut u64
      var $t2: &mut u64
      var $t3: &u64
+     var $t4: &u64
   0: $t0 := 22
   1: $t1 := borrow_local($t0)
   2: $t2 := m::f($t1)
   3: $t1 := infer($t2)
   4: $t3 := borrow_local($t0)
-  5: return ()
+  5: $t4 := infer($t3)
```

Comment:
Why do we have new copy here and elsewhere?

I'm generally good with the change but we need to be careful if the generated code structure changes, it can effect our hard work on optimizations in subsequent phases.

Reply:
Good catch.  Some might be caught by variable coalescing, but this one is not.

This is showing up some problem with `bytecode_generator`.  The only difference before conversion is that one pattern is a `Pattern::Wildcard`, while the other is a `Pattern::Var`.
```
            Pattern::Wildcard(id) => {
                let pat_ty = self.env().get_node_type(*id);
                if !match_mode.is_probing() {
                    let temp = self.new_temp(pat_ty);
                    self.emit_assign_with_convert(*id, temp, values[0], AssignKind::Inferred)
                }
            },
            Pattern::Var(var_id, sym) => {
                if match_mode.shall_bind_var(*sym) {
                    let local = self.find_local_for_pattern(*var_id, *sym, next_scope);
                    self.emit_assign_with_convert(id, local, values[0], AssignKind::Inferred)
                }
            },
```
I'm going to try to debug this.

Reply:
I've fixed the handling of Wildcard in 2 places to avoid an unnecessary assignment.  There may still be an unnecessary `freeze` inserted, but that should match the var behavior, I think.